lib/BMP280-arduino-library-master/BMP280.h

```diff
@@ -0,0 +1,135 @@
+/*
+	BMP280.h
+	Bosch BMP280 pressure sensor library for the Arduino microcontroller.
+	This library uses I2C connection.
+
+	Uses floating-point equations from BMP280 datasheet.
+
+	modified by mhafuzul islam
+
+	version 1.01		 16/9/2014 initial version
+	
+	Our example code uses the "pizza-eating" license. You can do anything
+	you like with this code. No really, anything. If you find it useful,
+	buy me italian pizza someday.
+*/
+
+#ifndef BMP280_h
+#define BMP280_h
+
+#if defined(ARDUINO) && ARDUINO >= 100
+#include "Arduino.h"
+#else
+#include "WProgram.h"
+#endif
+
+// #define _debugSerial
+// #define _debugTestData
+
+class BMP280
+{
+	public:
+		BMP280(); // base type
+
+		char begin();
+		char begin(int sdaPin, int sclPin);
+			// call pressure.begin() to initialize BMP280 before use
+			// returns 1 if success, 0 if failure (i2C connection problem.)
+				
+		short getOversampling(void);
+		char  setOversampling(short oss);
+		
+		char startMeasurment(void);
+			// command BMP280 to start a pressure measurement
+			// oversampling: 0 - 3 for oversampling value
+			// returns (number of ms to wait) for success, 0 for fail
+		
+		char calcTemperature(double &T, double &uT);
+			// calculation the true temperature from the given uncalibrated Temperature 
+			
+		char calcPressure(double &P, double uP);
```

Comment:
W jakich jednostkach zwracane jest ciśnienie??

Reply:
to jest dobre pytanie. Bardzo gorąco polecam wysłuchać tego fragmentu o templejtach z prezentacji [Why C++ Sails When the Vasa Sank - Scott Meyers](https://youtu.be/9vyh1APsMAE?t=1071) (17:50)

mozemy uzyc czegos podobnego

Reply:
Ile osi symetrii ma krowa?

Reply:
a w połączeniu z [User-defined literals](https://en.cppreference.com/w/cpp/language/user_literal) mozemy miec nawet cos czytelnego